lib/ecto/type.ex

```diff
@@ -449,27 +449,29 @@ defmodule Ecto.Type do
   end
 
   def load(:time, %Time{} = time, _loader) do
-    {:ok, zerofy_microsecond(time)}
+    {:ok, truncate_usec(time)}
   end
 
   def load(:time_usec, %Time{} = time, _loader) do
-    {:ok, time}
+    {:ok, pad_usec(time)}
   end
 
   def load(:naive_datetime, %NaiveDateTime{} = naive_datetime, _loader) do
-    {:ok, zerofy_microsecond(naive_datetime)}
+    {:ok, truncate_usec(naive_datetime)}
   end
 
   def load(:naive_datetime_usec, %NaiveDateTime{} = naive_datetime, _loader) do
-    {:ok, naive_datetime}
+    {:ok, pad_usec(naive_datetime)}
   end
 
   def load(:utc_datetime, %NaiveDateTime{} = naive_datetime, _loader) do
-    naive_datetime |> zerofy_microsecond |> DateTime.from_naive("Etc/UTC")
+    naive_datetime |> truncate_usec() |> DateTime.from_naive("Etc/UTC")
   end
 
   def load(:utc_datetime_usec, %NaiveDateTime{} = naive_datetime, _loader) do
-     DateTime.from_naive(naive_datetime, "Etc/UTC")
+    with {:ok, %DateTime{} = datetime} <- DateTime.from_naive(naive_datetime, "Etc/UTC") do
```

Comment:
This `with` might suggest that an error response might be forwarded from `DateTime.from_naive/2`. However, that function always returns `{:ok, datetime}`. 

In addition, I think the pattern match on the struct `%DateTime{}` is not needed in this case :)

What do you think then about?

```elixir
{:ok, naive_datetime |> DateTime.from_naive!("Etc/UTC") |> pad_usec()}

# or:

{:ok, pad_usec(DateTime.from_naive!(naive_datetime, "Etc/UTC"))}

# or:

datetime = DateTime.from_naive!(naive_datetime, "Etc/UTC")
{:ok, pad_usec(datetime)}

# or:

{:ok, datetime} = DateTime.from_naive(naive_datetime, "Etc/UTC")
{:ok, pad_usec(datetime)}
```

Reply:
good catch! I went with:

    naive_datetime |> pad_usec() |> DateTime.from_naive("Etc/UTC")

to get a nice symmetry with a clause above:

    naive_datetime |> truncate_usec() |> DateTime.from_naive("Etc/UTC")